src/main/scala/com/cognite/sdk/scala/v1/files.scala

```diff
@@ -23,6 +22,7 @@ final case class File(
     directory: Option[String] = None,
     source: Option[String] = None,
     externalId: Option[String] = None,
+    instanceId: Option[InstanceId] = None,
```

Comment:
couple thoughts
- will it trigger some changes in transformations (also in case it can be created or read but not updated)
- can a file have both externalId&instanceId :thinking: maybe nothing will break if it can, except for maybe that CogniteIdOrInstanceId handling?

Reply:
* It shouldn't but you are right I need to make absolutely sure
 * My understanding is that it can't. I was told the api spec which said you can update the externalId when updating by instance id was wrong, and I don't think there is another way to set the externalId.

Reply:
For point 1, I think this will be always none, since instance Id cannot be set at the same time as externalId as far as I understand and since we can only access files using external Id (so far.) so I think without changes in datasource it won't trigger any changes

Reply:
> will it trigger some changes in transformations (also in case it can be created or read but not updated)

Actually, it can't be created either, just read... creation goes through DMS API

> can a file have both externalId&instanceId 🤔 maybe nothing will break if it can, except for maybe that CogniteIdOrInstanceId handling?

Since creation and updates are only fully supported in the DMS API, I don't see this happening